scripts/build.sh

```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env bash
+set -e
+
+git clone https://github.com/opentripplanner/OpenTripPlanner.git otp || true
+
+cd otp
+git checkout b391945480c37eaffc881df427700ef351b4b19d
```

Comment:
__question__(non-blocking): do you know why the submodule wasn't working?

Reply:
Unfortunately not. I didn't want to keep wasting time on it to try to narrow down the exact change that resolved it, and I'm still not sure if it was related to the submodule or not. I started cloning directly (into a directory not named after the repo, which _really_ shouldn't make a difference) rather than using the submodule, and I stopped renaming and moving the jar before running the build/serve (though the dockerfile does still move the jar, it never had any issues). It was failing in a handful of fairly different ways every time it was run, sometimes during the `mvn clean package`, and sometimes during the OTP build, super frustrating to debug. They all seemed related to it not being able to find files, during the package, it would say that some random class was missing (when the file definitely existed in the submodule), and all the OTP build failures seemed to be caused by some config file (that I assume should exist inside the jar) being inaccessible, it usually prints out the OTP version, branch, and commit that the package is using, but it would just say UNKNOWN for everything.

Also, I really don't like having an arbitrary hash just in the middle of the script like this, and I should probably move it into an env var.

Reply:
Ok, moved the repo and commit into env vars, I had to do some funky stuff with .env and .envrc to get a single place that both regular local builds and local docker build pull from, then they're also defined in the github actions for being passed into AWS deploys. I feel a little weird about having the hash defined in two separate places, and now that I think about it, maybe I could get github actions to pull from `.env` also...

Reply:
From a quick glance it does seem that there are existing actions which source .env's https://github.com/marketplace/actions/dotenv-action

Reply:
> falti/dotenv-action@v1.0.4 is not allowed to be used in mbta/otp-deploy. Actions in this workflow must be: within a repository that belongs to your Enterprise account, created by GitHub, verified in the GitHub Marketplace, or matching the following: ...

🥲 Options are, figure out a way to do it with bash directly in our action, put a copy of the dotenv-action repo directly in the `.github/actions` dir (ick), make an MBTA fork, or get it added to the allowlist. Adding it to the allowlist is probably easiest but I'm not sure how much red tape there is to do that, I'll ask.

Reply:
Alternatively, you could convert the `.env` into `set-otp-repo.sh` (and convert the env declarations to `export`'s) and use `source_env set-otp-repo.sh`(https://direnv.net/man/direnv-stdlib.1.html#codesourceenv-ltfileordirpathgtcode) inside of the `.envrc`

Then you could call the `set-otp-repo.sh` script which exports the variables in the workflow?

Reply:
I somewhat retract my previous endorsement of the dotenv-action and instead think that the above is probably a better option, given it's less external code and it's relatively simple to do it via bash.

Reply:
The main reason I'm using a `.env` file at all is because it's compatible with docker-compose, but I don't think it's actually using it the way that I expected, and the args are just defined from the environment it's being run in, so hopefully `.envrc` alone would cover it. I'll give that method a shot.

Reply:
Run is still ongoing, but it's past the point of using the env vars and it looks like it's working 🥳 
I made `.envrc.global` to hold the global vars, which can be sourced normally in the action, and an `.envrc.local.template` which can be created to override the values, `.envrc` just pulls from both. Thanks for the advice, I'm happy with this approach.

Reply:
Cool, I'm glad that worked! I think it turned out well!